action.yml

```diff
@@ -234,6 +239,19 @@ runs:
           fi
         fi
 
+    - uses: actions/github-script@v7
+      id: requirements-hash
+      with:
+        script: return require('fs').createReadStream(require('path').join(process.env.GITHUB_ACTION_PATH, 'requirements.txt')).pipe(require('crypto').createHash('sha1').setEncoding('hex'), 'finish').digest('hex')
+        result-encoding: string
+
+    - uses: actions/cache@v4
+      if: ${{ inputs.cache == 'true' }}
+      with:
+        key: venv-${{ runner.os }}-${{ steps.setup-python.outputs.python-version }}-${{ steps.requirements-hash.outputs.result }}
```

Comment:
I'm not a big fan of using another action just to get the hash of requirements.txt. Especially when the other action can be used very maliciously.

According to the docs, we can't just [`hashfiles(env.GITHUB_ACTION_PATH + '/requirements.txt')`](https://docs.github.com/en/actions/learn-github-actions/expressions#hashfiles) because that path is not within the `GITHUB_WORKSPACE`.

Still, I think this can be done more securely using native shells.

Reply:
We could just copy over the requirements.txt file to the .cpp-linter_cache/ folder in `repo-root`. Then we could use the `hashfiles('.cpp-linter_cache/requirements.txt')`

Reply:
rewrote using the python shell